tempesta_fw/t/unit/test_tfw_str.c

```diff
@@ -101,102 +102,130 @@ TEST(tfw_strcpy, dst_compound)
-	TFW_STR(s1, "abcdefghijklmnop");
-	TFW_STR(s2, "0123456789");
+	TfwStr *s1 = make_compound_str2("abc", "defgh");
+	TfwStr *s2 = make_compound_str2("01234", "56789");
 
 	chunks1 = TFW_STR_CHUNKN(s1);
 	chunks2 = TFW_STR_CHUNKN(s2);
-
+	TFW_DBG("196:s1:d:%s;chnks:%d\n", s1->data, s2->chunknum);
 	EXPECT_ZERO(tfw_strcat(str_pool, s1, s2));
 	EXPECT_TRUE(TFW_STR_CHUNKN(s1) == chunks1 + chunks2);
-	EXPECT_TRUE(tfw_str_eq_cstr(s1, "abcdefghijklmnop0123456789",
-				    sizeof("abcdefghijklmnop0123456789") - 1,
+	EXPECT_TRUE(tfw_str_eq_cstr(s1, "abcdefgh0123456789",
+				    sizeof("abcdefgh0123456789") - 1,
 				    0));
 }
 
 TEST(tfw_stricmpspn, returns_true_only_for_equal_tfw_strs)
 {
-	TFW_STR(s1, "abcdefghijklmnopqrst");
-	TFW_STR(s2, "ABcDefGHIJKLmnopqrst");
-	TFW_STR(s3, "abcdefghi");
-	TFW_STR(s4, "abcdefghijklmnopqrst_the_tail");
-
-	EXPECT_TRUE(tfw_stricmpspn(s1, s2, 0) == 0);
-	EXPECT_FALSE(tfw_stricmpspn(s1, s3, 0) == 0);
-	EXPECT_TRUE(tfw_stricmpspn(s1, s3, 'f') == 0);
-	EXPECT_FALSE(tfw_stricmpspn(s1, s4, 0) == 0);
-	EXPECT_TRUE(tfw_stricmpspn(s1, s4, 't') == 0);
+	DEFINE_TFW_STR(s1, "abcdefghijklmnopqrst");
+	DEFINE_TFW_STR(s2, "ABcDefGHIJKLmnopqrst");
+	DEFINE_TFW_STR(s3, "abcdefghi");
+	DEFINE_TFW_STR(s4, "abcdefghijklmnopqrst_the_tail");
+	TFW_DBG("ret_true:start_assert\n");
+	EXPECT_TRUE(tfw_stricmpspn(&s1, &s2, 1) == 0);
+	TFW_DBG("ret_true:first - ok\n");
+	EXPECT_FALSE(tfw_stricmpspn(&s1, &s3, 1) == 0);
+TFW_DBG("ret_true: second - ok\n");
+	EXPECT_TRUE(tfw_stricmpspn(&s1, &s3, 'f') == 0);
+	TFW_DBG("ret_true:third - ok\n");
+	EXPECT_FALSE(tfw_stricmpspn(&s1, &s4, 0) == 0);
+	TFW_DBG("ret_true:fourth - ok\n");
+	EXPECT_TRUE(tfw_stricmpspn(&s1, &s4, 't') == 0);
 }
 
 TEST(tfw_stricmpspn, handles_plain_and_compound_strs)
 {
+	int res;
 	TfwStr s1 = {
 		.len	= sizeof("abcdefghijklmnopqrst") - 1,
-		.ptr	= "abcdefghijklmnopqrst"
+		.data	= "abcdefghijklmnopqrst"
 	};
-	TFW_STR(s2, "abcdefghijklmnopqrst");
-	TFW_STR(s3, "abcdefghi");
-	TFW_STR(s4, "abcdefghijklmnopqrst_the_tail");
-
-	EXPECT_TRUE(tfw_stricmpspn(&s1, s2, 0) == 0);
-	EXPECT_FALSE(tfw_stricmpspn(&s1, s3, 0) == 0);
-	EXPECT_TRUE(tfw_stricmpspn(&s1, s3, 'f') == 0);
-	EXPECT_FALSE(tfw_stricmpspn(&s1, s3, 'z') == 0);
-	EXPECT_FALSE(tfw_stricmpspn(&s1, s4, 0) == 0);
-	EXPECT_TRUE(tfw_stricmpspn(&s1, s4, 't') == 0);
+	DEFINE_TFW_STR(s2, "abcdefghijklmnopqrst");
+	DEFINE_TFW_STR(s3, "abcdefghi");
+	DEFINE_TFW_STR(s4, "abcdefghijklmnopqrst_the_tail");
+	res = tfw_stricmpspn(&s1, &s2, 0);
+	TFW_DBG("cmpspn:185:res:%d;l1:%lu;l2:%lu\n", res, s1.len, s2.len);
+	EXPECT_TRUE(tfw_stricmpspn(&s1, &s2, 0) == 0);
+	EXPECT_FALSE(tfw_stricmpspn(&s1, &s3, 0) == 0);
+	EXPECT_TRUE(tfw_stricmpspn(&s1, &s3, 'f') == 0);
+	EXPECT_FALSE(tfw_stricmpspn(&s1, &s3, 'z') == 0);
+	EXPECT_FALSE(tfw_stricmpspn(&s1, &s4, 0) == 0);
+	EXPECT_TRUE(tfw_stricmpspn(&s1, &s4, 't') == 0);
 }
 
 TEST(tfw_stricmpspn, handles_empty_strs)
 {
 	TfwStr s1 = {
 		.len	= 0,
-		.ptr	= "garbage"
+		.data	= "garbage"
```

Comment:
`{ .data = "xxxxx" }` is sufficient here. All other members will be initialized to zero values.